escodegen.js

```diff
@@ -1176,15 +1176,19 @@
             withIndent(function () {
                 var guard;
 
-                result = [
-                    'catch' + space + '(',
-                    that.generateExpression(stmt.param, Precedence.Sequence, E_TTT),
-                    ')'
-                ];
+                if (!stmt.param) {
```

Comment:
Always use a positive condition when there's an alternative branch. Drop the `!` and swap the branches.